src/Emu/Metadata/SupportFiles/FrontierLabs/LogFile.cs

```diff
@@ -354,47 +401,39 @@ public override bool ExtractInformation()
         {
             using (StreamReader reader = new StreamReader(this.FilePath))
             {
-                var headerData = HeaderParser(reader);
-
-                if (!headerData.IsLogFile)
-                {
-                    return false;
-                }
-
-                this.Sensor = new Sensor() with
-                {
-                    Firmware = headerData.Firmware,
-                    SerialNumber = headerData.SerialNumber,
-                    PowerSource = headerData.PowerSource,
-                };
-
                 string line;
-                int itemNumber = 0;
 
                 while ((line = reader.ReadLine()) != null)
                 {
-                    if (line.Contains(SDCardString))
+                    if (line.Contains(FrontierLabsLogString))
+                    {
+                        this.SensorLogs.Add(SensorParser(reader));
+                    }
+                    else if (line.Contains(SDCardString))
                     {
-                        this.MemoryCardLogs.Add((MemoryCardParser(reader), itemNumber));
-                        itemNumber++;
+                        this.MemoryCardLogs.Add(MemoryCardParser(reader, line));
                     }
                     else if (line.Contains(RecordingString))
                     {
-                        this.RecordingLogs.Add(RecordingParser(reader, line, itemNumber));
-                        itemNumber++;
+                        this.RecordingLogs.Add(RecordingParser(reader, line));
                     }
                     else if (line.Contains(LocationString))
                     {
-                        this.Location = LocationParser(line.Split(LocationString).Last());
+                        this.LocationLogs.Add(LocationParser(line));
                     }
                 }
             }
 
+            this.SensorLogs = this.SensorLogs.OrderBy(log => log.TimeStamp).ToList();
+            this.MemoryCardLogs = this.MemoryCardLogs.OrderBy(log => log.TimeStamp).ToList();
+            this.RecordingLogs = this.RecordingLogs.OrderBy(log => log.TimeStamp).ToList();
+            this.LocationLogs = this.LocationLogs.OrderBy(log => log.TimeStamp).ToList();
+
             return true;
         }
 
-        public record RecordingRecord(string Name, int ItemNumber, double? BatteryLevel, double? Voltage, Microphone[] Microphones);
+        public record RecordingRecord(string Name, double? BatteryLevel, double? Voltage, Microphone[] Microphones, LocalDateTime TimeStamp);
 
-        public record FileHeader(bool IsLogFile, string Firmware, string SerialNumber, string PowerSource);
+        public record DataRecord(object Data, LocalDateTime TimeStamp);
```

Comment:
```suggestion
        public record DataRecord<T>(T Data, LocalDateTime TimeStamp);
```

Did you try making this generic? It should be more performant and safer from a typing perspective.